flake8_trio.py

```diff
@@ -1441,9 +1467,131 @@ def visit_AsyncFunctionDef(
 
     def visit_Call(self, node: ast.Call):
         if self.async_function and not getattr(node, "awaited", False):
-            blocking_calls = self.options.trio200_blocking_calls
-            if key := fnmatch_qualified_name([node.func], *blocking_calls):
-                self.error(node, key, blocking_calls[key])
+            self.visit_blocking_call(node)
+
+    def visit_blocking_call(self, node: ast.Call):
+        blocking_calls = self.options.trio200_blocking_calls
+        if key := fnmatch_qualified_name([node.func], *blocking_calls):
+            self.error(node, key, blocking_calls[key])
+
+
+@error_class
+class Visitor21X(Visitor200):
+    error_codes = {
+        "TRIO210": "Sync HTTP call {} in async function should use httpx.AsyncClient",
+        "TRIO211": (
+            "Likely sync HTTP call {} in async function should use httpx.AsyncClient"
+        ),
+    }
+
+    def __init__(self, *args: Any, **kwargs: Any):
+        super().__init__(*args, **kwargs)
+        self.imports: set[str] = set()
+
+    def visit_ImportFrom(self, node: ast.ImportFrom):
+        if node.module in ("httpx", "urllib3"):
+            self.imports.add(node.module)
+
+    def visit_Import(self, node: ast.Import):
+        for name in node.names:
+            if name.name in ("httpx", "urllib3"):
+                # Could also save the name.asname for matching
+                self.imports.add(name.name)
+
+    def visit_blocking_call(self, node: ast.Call):
+        http_methods = {
+            "get",
+            "options",
+            "head",
+            "post",
+            "put",
+            "patch",
+            "delete",
+        }
+        func_name = ast.unparse(node.func)
+        for http_package in "requests", "httpx":
+            if get_matching_call(node, *http_methods | {"request"}, base=http_package):
+                self.error(node, func_name, error_code="TRIO210")
+                return
+
+        if func_name in (
+            "urllib3.request",
+            "urllib.request.urlopen",
+            "request.urlopen",
+            "urlopen",
+        ):
+            self.error(node, func_name, error_code="TRIO210")
+
+        elif (
+            "urllib3" in self.imports
+            and isinstance(node.func, ast.Attribute)
+            and node.func.attr == "request"
+            and node.args
+            and isinstance(node.args[0], ast.Constant)
+            and isinstance(node.args[0].value, str)
+            and node.args[0].value.lower() in http_methods | {"trace", "connect"}
+        ) or (
+            "httpx" in self.imports
+            and isinstance(node.func, ast.Attribute)
+            and node.func.attr in http_methods | {"request", "stream", "send"}
```

Comment:
I think this is probably too false-alarm-prone; without the extra evidence from the first argument being a known constant we'll error on *any* method with one of these extremely common names.  For example:

```python
import httpx

some_dict.get(...)  # should not raise a TRIO warning!
```

Reply:
It still has to be inside an async function, and not be awaited. But yeah could also track the variable name a call to `httpx.Client()` is saved to (and do the same with urllib3), as is currently done with nurseries.

Removed it for now, and bumped version.

Reply:
> It still has to be inside an async function, and not be awaited. But yeah could also track the variable name a call to httpx.Client() is saved to (and do the same with urllib3), as is currently done with nurseries.

Probably a good follow-up PR once we get the other parts of #58 done 👍